### PR TITLE
More zsh history options

### PR DESCRIPTION
**Pull request by MasseR**
Configuration options for `HIST_EXPIRE_DUPS_FIRST` and `EXTENDED_HISTORY`

### CODE REVIEW

**Comment by uvNikita** [comment]
Thanks for the PR!

Looks good to me, the only suggestion I have is to rename `zsh.history.extendedHistory` option to just `zsh.history.extended`, since its clear that the option configures history.

And could you please change the commit message to something like: `zsh: add extended, expireDupsFirst history options` for it to be alligned with our [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/CONTRIBUTING.md) guidelines?

**Comment by MasseR** [comment]
Sure thing. I can get back to this hopefully later today.

**Comment by MasseR** [comment]
- [X] rename `extendedHistory` to `extended`
- [X] rename commit to `zsh: Add extended, expierDups history options`

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -50,6 +50,18 @@ let
         '';
       };
 
+      expireDupsFirst = mkOption {
```
Comment (rycee) [inline modules/programs/zsh.nix]: Could simply call it `expireDuplicatesFirst`, it's not a huge length difference 🙂

**Comment by MasseR** [comment]
Renamed to `expireDuplicatesFirst`

**Comment by rycee** [comment]
Thanks! I've rebased to master in 06a984e4ff62b3f07e7e41ff93b45f0b759c9c56. I changed the commit message slightly to have a lowercase a in add 🙂

**Pull request closed without merging**